src/types/utils.cpp

```diff
@@ -428,6 +428,90 @@ catch (...)
     return {};
 }
 
+// Routine Description:
+// - Pre-process text pasted (presumably from the clipboard) with provided option.
+// Arguments:
+// - wstr - String to process.
+// - option - option to use.
+// Return Value:
+// - The result string.
+std::wstring Utils::FilterStringForPaste(const std::wstring_view wstr, const FilterOption option)
+{
+    std::wstring filtered;
+    filtered.reserve(wstr.length());
+
+    const auto isControlCode = [](wchar_t c) {
+        if (c >= L'\x20' && c <= L'\x7f')
+        {
+            // Printable ASCII characters + DEL.
+            return false;
+        }
+
+        if (c > L'\x7f')
+        {
+            // Not a control code.
+            return false;
+        }

```

Comment:
What about the C1 control codes? Shouldn't we be filtering those out too? I haven't tested this on other clients, but it occurs to me that an app that supports 8-bit mode may well interpret C1 control codes even if the mode isn't enabled. That may then allow the paste content to break out of the bracket, or send through keystrokes that we were intending to filter.

Reply:
We probably should, yet. I had a concern about breaking UTF-8 code units, but realized that this is _after_ transformation to UTF-16.

Reply:
C1 control codes added!